versions/1/data_stream/fields/fields.spec.yml

```diff
@@ -6,66 +6,82 @@ spec:
   type: array
   items:
     type: object
+    # TODO: We should disable additionalProperties. There are properties used in
+    # elastic/integrations that have no purpose or are typos.
     additionalProperties: true
     properties:
       name:
-        description: Name of field
-        type: string
-      title:
-        description: Title of field
+        description: >
+          Name of field. Names containing dots are automatically split into
+          sub-fields.
         type: string
+        # TODO: Should we allow wildcards if they are not implemented in Fleet?
+        pattern: '^[_@A-Za-z0-9]+(\.[_@A-Za-z0-9]+)*$'
```

Comment:
Yes, I think we need to allow wildcards.

The reason we don't see more wildcards now is because we are interpreting objects without wildcards as if the wildcard would be at the end by convention. For example for `labels`, the matching path pattern would be `labels.*`. See how the wildcard is added also in beats [here](https://github.com/elastic/beats/blob/ea207346d651448b8917b0791b2b117b9f9b9212/libbeat/template/processor.go#L471-L473).
There are cases where the `*` needs to be in the middle of a path, like in the [prometheus cases](https://github.com/elastic/integrations/blob/b0f94db6310604548ca98c322fa4dd5489106a0a/packages/prometheus/data_stream/collector/fields/fields.yml#L15), and they cannot benefit of the previous convention.

An alternative would be to have an explicit field to define the [`path_match`](https://www.elastic.co/guide/en/elasticsearch/reference/8.1/dynamic-templates.html#path-match-unmatch), but this could be error prone (typos between the name of the field and its matching path), would need changes in fields validation, and it would break packages that currently use the wildcard.

Reply:
I amended the pattern to allow `*` and added a test case with a `foo.*`. I'll create a follow-up issue to continue discussion of the design for dynamic mappings.